BEST_PRACTICE.md

```diff
@@ -0,0 +1,247 @@
+
+# your output file
+TARGET = project-name
+
+# your project name
+TARGET = your-project
+
+CFLAGS = -Ideps -Wall
+
+# all the source files
+SRC = $(wildcard src/*.c)
+SRC += $(wildcard deps/*/*.c)
+
+OBJS = $(SRC:.c=.o)
+
+.PHONY:
+all: $(TARGET)
+
+.PHONY:
+$(TARGET): $(OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $(TARGET) $(OBJS)
+
+.PHONY:
+%.o: %.c
+	$(CC) $(DEP_FLAG) $(CFLAGS) $(LDFLAGS) -o $@ -c $<
+
+.PHONY:
+clean:
+	rm -f $(OBJS)
+
+.PHONY:
+install: $(TARGET)
+	cp -f $(TARGET) $(PREFIX)
+
+.PHONY:
+uninstall: $(PREFIX)/$(TARGET)
+	rm -f $(PREFIX)/$(TARGET)
+```
+
+This is a basic Makefile, and should work for most of your projects.
+
+You *could* have your Makefile install the libraries upon running it, but you
+would only need to do that to get the latest version of the library(s), in this
+case you proboly don't want that. You typicly want yo get the lates stable version
+for that library. By having a `package.json` file in your project repo, you can
+specify what packages you need, and what version of that package. Now have a look
+at a example `package.json` file for your project: (executable package)
+
+### Example package.json for exacutable project
+
+```
+{
+  "name": "executable-name",
+  "version": "1.0.0",
+  "repo": "your-github-name/project-name",
+  "dependencies": {
+    "stephenmathieson/trim.c": "0.0.2",
+    "clibs/commander": "1.3.2",
+    "clibs/logger": "0.0.1",
+  },
+  "install": "make install",
+  "uninstall": "make uninstall"
+}
+```
+
+Starting from the top, `"name"` is your package name. `"version"` is your package version.
+`"repo"` is the location of your project, (not including the `https://github.com/`).
+`"dependencies"` is all the dependencies your project requiers, along with there version.
+`"install"` is the command to install your program (ran as root), (tip: if your project
+requiers more then one command to install it, like need to run `./configure`, before `make`,
+then do this: `"install": "./configure && make && make install"`). `"uninstall"` is the command
+to uninstall your project, [more on that later](#install-and-uninstall-executables).
+
+***NOTE:** Make sure you have a relese as the same version in your `package.json` file, otherwise
+the download will fail. If you allwasy want your package at the latest version, then put `master`
+as your version.*
+
+## Making your own libraries
+
+Now that your know how to use libraries, heres how to make your own:
+
+Like before, heres a typical project directory tree:
+
+```
+your-library-c/
+│
+├── deps/
+│   ├── path-join.c/
+│   │   ├── path-join.h
+│   │   ├── path-join.c
+│   │   └── package.json
+│   │
+│   └── strdup/
+│       ├─ strdup.h
+│       ├─ strdup.c
+│       └─ package.json
+│
+├── LICENSE
+│
+├── Makefile
+│
+├── README.md
+│
+├── package.json
+│
+├── src/
+│   ├── library.c
+│   ├── library.h
+│   └── etc...
+│
+└── test.sh
+```
+
+Also like before, your have a `deps` directory (depending on your library, you may not need any
```

Comment:
Why is this bolded?

Reply:
Its not, Github is not printing it correctly with all the `***` from above. I'll try to fix that...